src/Features/Core/Portable/SolutionCrawler/SolutionCrawlerRegistrationService.cs

```diff
@@ -91,7 +91,10 @@ public void Reanalyze(Workspace workspace, IIncrementalAnalyzer analyzer, IEnume
                 var coordinator = default(WorkCoordinator);
                 if (!_documentWorkCoordinatorMap.TryGetValue(workspace, out coordinator))
                 {
-                    throw new ArgumentException("workspace");
+                    // this can happen if solution crawler is already unregistered from workspace.
+                    // one of those example will be VS shutting down so host workspace is disposed but there is a pending
```

Comment:
ya, I should have said Roslyn Package is disposed.

http://index/#Microsoft.VisualStudio.LanguageServices/Implementation/LanguageService/AbstractPackage%25602.cs,126